src/uptime.rs

```diff
@@ -0,0 +1,131 @@
+use std::fmt;
+use std::fs::read_to_string;
+
+use serde::{Deserialize, Serialize};
+
+use crate::Config;
+use crate::Module;
+
+/// /proc/uptime contains two values, the first one represents the time the
+/// system has been on (in seconds) and the second value is the sum of how much
+/// time each core has spent idle.
+const UPTIME_PATH: &str = "/proc/uptime";
+
+#[derive(Serialize, Deserialize)]
+#[serde(default)]
+pub struct Uptime {
+    header: String,
+
+    #[serde(skip)]
+    /// Uptime in seconds
+    uptime: u32,
+
+    /// When true, uptime will be outputted like: 2d 10h 3m
+    short_output: bool,
+}
+
+struct Duration {
+    days: u32,
+    hours: u32,
+    mins: u32,
+    secs: u32,
+}
+
+impl Duration {
+    fn from_secs(secs: u32) -> Self {
+        Self {
+            days: secs / 60 / 60 / 24,
+            hours: secs / 60 / 60 % 24,
+            mins: secs / 60 % 60,
+            secs,
+        }
+    }
+}
+
+impl fmt::Display for Duration {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let mut result = String::new();
+
+        if self.days > 0 {
+            result.push_str(&format!("{} days, ", self.days));
+        }
+        if self.hours > 0 {
+            result.push_str(&format!("{} hours, ", self.hours));
+        }
+        if self.mins > 0 {
+            result.push_str(&format!("{} mins", self.mins));
+        }
+        if self.days == 0 && self.hours == 0 && self.mins == 0 {
+            result = format!("{} secs", self.secs);
+        }
+        // In case duration is exactly some amount of days or hours
+        result = result
+            .trim_end_matches(|c: char| c == ',' || c.is_whitespace())
+            .to_string();
+
+        write!(f, "{}", result)
+    }
+}
```

Comment:
i would suggest we avoid using `fmt::Display` here and instead implement functions for the struct to format these correctly, an idea would be:
```
fn num_days(&self) -> Option<String> {
    match self.days {
        0 -> None // maybe possible to return String::empty here and skip the option
        1 -> Some(String::from("{} day", 1))
        _ -> Some(String::from("{} days", self.days))
    }
}

fn to_string(&self, short: bool) -> String {
    let mut uptime = String::new();
    uptime.push_str(self.num_days().unwrap_or_else(String::Empty));

    if uptime.empty() {
        uptime.push_str(self.num_secs());
    }
}
```